frontend/src/components/Banner.js

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2021 Christoph Ehm <christoph.ehmendoerfer@campus.tu-berlin.de>
+
+import React from "react";
+import { useMediaQuery } from "react-responsive";
+
+import HeaderTypography from "./HeaderTypography";
+import ParagraphTypography from "./ParagraphTypography";
+import { Link } from "react-router-dom";
+import RoundedEdgesButton from "./RoundedEdgesButton";
+
+import { textColor, activeTextColor } from "../assets/jss/colorPalette";
+import bannerImage from "../assets/img/banner.png";
+
+/**
+ * Displays a welcoming background image with a button.
+ * @param user user object of the currently logged in account
+ * @returns {JSX.Element}
+ */
+export default function WelcomeBanner({ user }) {
+	const welcomeTitle = "Welcome to NFT The World!";
+	const welcomeSubtitle = <>
+		Breathtaking NFT shopping lounge.<br/>
+		Own your virtual map area from Nuremburg and Riga!
+	</>;
+	return <Banner user={user} title={welcomeTitle} subtitle={ welcomeSubtitle }/>;
+}
+
+/**
+ * Displays background banner with headline "title", text "subtitle" and login-dependent button.
+ * @param user user object of the currently logged in account
+ * @returns {JSX.Element}
+ */
+export function Banner({ subtitle, title, user }) {
+	const render=() => (
+		<div style={containerStyle}>
+			<div style={backgroundImageStyle}>
+				<div style={{ ...bannerItemStyle, marginBottom: "11px" }}>{ bannerTitle }</div>
+				<div style={{ ...bannerItemStyle, height: "66px", marginBottom: "32px" }}>{ bannerSubtitle }</div>
+				<div style={{ ...bannerItemStyle, }}>{ button }</div>
+			</div>
+		</div>
+	);
+
+	const largeSizeThreshold = 900;
+	const isLarge = useMediaQuery({ query: `(min-width: ${largeSizeThreshold}px)` });
+
+	const bannerTitle = <BannerTitle isLarge={isLarge}>{ title }</BannerTitle>;
+	const bannerSubtitle = <BannerSubTitle isLarge={isLarge}>{ subtitle }</BannerSubTitle>;
+	const button = <BannerButton isLoggedIn={(user != null)} isLarge={isLarge}/>;
+
+	const containerStyle = ({ minHeight: "305px", });
+	const backgroundImageStyle = ({
+		padding: "44px",
+		alignItems: "center",
+		justifyContent: "center",
+		textAlign: "center",
+	
```

Comment:
maybe remove empty line and maybe make it a one liner?

Reply:
Haha, this is now a long line. At the beginning I wasn't sure whether I should do this but this is consistent :-D .

All those properties would otherwise take a lot of lines and are usually not-so-interesting details. By writing them on one line it's like collapsed.